### ISSUE TITLE
Error deploying udpipe2 trained model

### ISSUE THREAD

**Issue by LilitKharatyan** [ufal/udpipe#182]
I have trained upipe2 model following the given recommendations and guidelines. Now I have the weights and I try to use the model like this:
`python udpipe2_server.py 8000 Caval_model Caval_model C:\model_path XCL none`

but when I try to use the model like this:
`python udpipe2_client.py --service http://localhost:8000 --model Caval_model --input conllu --output conllu --tokenizer XCL --tagger Caval_model --parser Caval_model C:\Users\User\Downloads\sample.conllu`

I receive the following error:

Traceback (most recent call last):
  File "udpipe2_server.py", line 329, in do_GET
    output = model.predict(batch, tag, parse, writer)
  File "udpipe2_server.py", line 139, in predict
    wembeddings = self._server_args.wembedding_server.compute_embeddings(self._network.args.wembedding_model, wembedding_input)
  File "C:\Users\User\PycharmProjects\CAVaL\udpipe\wembedding_service\wembeddings\wembeddings.py", line 110, in compute_embeddings
    model.load()
  File "C:\Users\User\PycharmProjects\CAVaL\udpipe\wembedding_service\wembeddings\wembeddings.py", line 52, in load
    self._transformers_model = transformers.TFAutoModel.from_pretrained(
  File "C:\Users\User\PycharmProjects\CAVaL\venv\lib\site-packages\transformers\utils\import_utils.py", line 1026, in __getattribute__
    requires_backends(cls, cls._backends)
  File "C:\Users\User\PycharmProjects\CAVaL\venv\lib\site-packages\transformers\utils\import_utils.py", line 1014, in requires_backends
    raise ImportError("".join(failed))
ImportError:
TFAutoModel requires the TensorFlow library but it was not found in your environment. Checkout the instructions on the
installation page: https://www.tensorflow.org/install and follow the ones that match your environment.
Please note that you may need to restart your runtime after installation.

udpipe requires Tensorflow 1 while embeddings - TensorFlow2. 
Can you please hint me if I am doing everything correctly or not? I will be grateful for help to use my model.

**Comment by foxik** [comment]
Hi, the error on line 361 indicates, that:
- during training you did not provide (=precompute) the contextualized embeddings (see the step 1 in Training New Models in https://github.com/ufal/udpipe/tree/udpipe-2)
- however, the `udpipe2_server.py` assumes you trained with contextualized embeddings, computes embeddings described by the `--wembedding_model` option of the trained model [which is by default `bert-base-multilingual-uncased-last4`], and passes these embeddings to the trained model -- which crashes, because it was trained without them

You have two ways to resolve the problem:
1. train a new model with contextualized embeddings. That will improve the performance of the model (assuming the language of your data is included in mBERT model). The embeddings need to be precomputed using the wembeddings script (as mentioned in the readme); the command would look something like
   ```sh
   python wembedding_service/compute_wembeddings.py --format=conllu PATH_TO_YOUR_TRAIN.conllu PATH_TO_YOUR_TRAIN.conllu.npz --model=bert-base-multilingual-uncased-last4
   ```
   and you need to run it also for dev/test data. (We precompute it, because there is no need to compute it 60 times during the training itself [60 is the default epoch number].)

   During training, the `udpipe2.py` script locates the precomputed embeddings by appending `.npz` to the train/dev/test conllu path name.
2. update `udpipe2_server.py` not to compute contextualized embeddings. You could do it by replacing the line https://github.com/ufal/udpipe/blob/308a7ccd63cb238326fdb0960e2d7f5acfd92813/udpipe2_server.py#L139 by `wembeddings = []`. (I have not really tried it, but I think it should work.)

The solution 2. will run faster, but will result in worse accuracy. But if you would prefer it, I could update `udpipe2_server.py` so that if the `--wembeddings_model` is empty, it would not compute the embeddings (and you could update `options.json` of you train model so that this option is empty). That would allow your model to run without modification of `udpipe2_server.py`. But please first try it yourself by modifying the `udpipe2_server.py` yourself, verifying that the model works for you, and that you are happy with the accuracy (because without mBERT, it will be several percentage points worse).

**Comment by LilitKharatyan** [comment]
Thank you very much for your response.
1. So the language of the model is classical Armenian (XCL) which to the best of my knowledge is not within the list of the languages supported by the mBERT model.
2. I tried to modify the server file the way you mentioned as well as removed the embeddings option from the json file, still, I received exactly the same error from the udpipe2.py file 361 line. 
3. I would be very thankful for the assistance because apparently (according to the training results) the model yields great results (way better than udpipe1) and I am kind of unable to use the model. 

Thanks again!

**Comment by foxik** [comment]
OK, if you do not want mBERT, lets update `udpipe2_server.py` to handle this case. Could you please attach here the following files from the trained model (feel free to zip them and attach a zip, or attach them one by one):
- log
- mappings.pickle
- options.json (the original one in the best case)
- and the tokenizer file (BTW, did you train it yourself?)

(I do not need the weights themselves, which are large.)

I will try to modify the `udpipe2_server.py` to handle the no-mBERT case, and tell you how to update the `options.json`.

**Comment by LilitKharatyan** [comment]
Thanks a lot. 
I am attaching all the files mentioned. 
The tokenizer is trained by myself, following the guidelines on the UDpipe2 page.
Thank you!
[xcl_model.zip](https://github.com/ufal/udpipe/files/13650163/xcl_model.zip)

**Comment by foxik** [comment]
Hi,

I have just committed d4be1239b2 allowing to run a model without contextualized embeddings. If you:
1. update to the latest commit in `udpipe-2` branch,
2. change the
   ```json
   "wembedding_model": "bert-base-multilingual-uncased-last4", 
   ```
   part in the `options.json` to
   ```json
   "wembedding_model": "",
   ```

then prediction with your model should work (it worked for me, when I used it with random weights instead of the trained ones).

Cheers!
